pkg/updatestrategy/updatestrategy.go

```diff
@@ -108,6 +108,10 @@ func (n *ProfileNodePoolProvisioner) Scale(ctx context.Context, nodePool *api.No
 }
 
 // Terminate terminates a node pool using the default provisioner.
-func (n *ProfileNodePoolProvisioner) Terminate(ctx context.Context, node *Node, decrementDesired bool) error {
-	return n.defaultProvisioner.Terminate(ctx, node, decrementDesired)
+func (n *ProfileNodePoolProvisioner) Terminate(ctx context.Context, nodePool *api.NodePool, node *Node, decrementDesired bool) error {
+	if provisioner, ok := n.profileMapping[nodePool.Profile]; ok {
+		return provisioner.Terminate(ctx, nodePool, node, decrementDesired)
+	}
+
+	return n.defaultProvisioner.Terminate(ctx, nodePool, node, decrementDesired)
```

Comment:
@mikkeloscar 
all the methods of this `ProfileNodePoolProvisioner` rely on the node pool to decide the (karpenter vs ASG) logic. this was the only method that always defaults to ASG and was not supporting the split logic.
I thought passing the nodePool would be consistent with the interface of the rest of the methods

Reply:
the ASG terminate logic relies on finding the ASG and decrement the number of desired nodes, ...etc and thus it will fail.

I can't modify the `ProfileNodePoolProvisioner` interface without `EC2NodePoolBackend` as they share the same interface

Reply:
I see, I was the one who introduced this pattern :see_no_evil: https://github.com/zalando-incubator/cluster-lifecycle-manager/pull/316

Ideally we can refactor it later to use types instead of a profileMapping map... Doesn't make sense to refactor in this PR ofc.